Cargo.toml

```diff
@@ -8,6 +8,7 @@ homepage = "http://github.com/DaneSlattery/hx711"
 description = "A no-std embedded-hal package for the hx711 load cell, based on the embedded hal. Has optional support for interrupts on the esp32."
 
 [dependencies]
+critical-section = "1.1.2"
```

Comment:
Can you remove the critical section in the dev dependencies and update the example to remove the critical section code from "main"?

Reply:
I haven't looked into the interrupt extension yet and I'm not sure how the change I made affects the example. Is the Mutex still required to read from both the interrupt function and main? I'd feel more comfortable if you could update the example. In my application I use esp-idf which is basically FreeRTOS and is a bit different to what you have in the examples.

Reply:
I will update the example, but perhaps it would be nifty to include how you are doing this in esp-idf?

Reply:
I'm currently trying to debug this as the critical section and interrupts are handled differently in esp-idf. I even think that if I want to do interrupts in esp-idf I'll have to modify this library to use the idf interrupts stuff (behind a feature flag). According to the nice folks on the esp-rs matrix chat, esp-hal interrupts are not compatible with esp-idf.